src/System.Console/src/System/IO/SyncTextReader.cs

```diff
@@ -31,8 +31,8 @@ protected override void Dispose(bool disposing)
             {
                 lock (_methodLock)
                 {
-                    // Explicitly pick up a potentially methodimpl'ed Dispose
-                    ((IDisposable)_in).Dispose();
+                    // This will always pick up TextReader.Dispose
```

Comment:
@James-Ko At the time the code was written, TextWriter and TextReader explicitly implemented IDisposable. Theoretically, also I can have my own TextReader/TextWriter implement IDisposable again, and completely skip the base's implementation. Such a class would experience a behavior change in above. However, the likelihood of such a thing is probably pretty low and such an implementation would have different broken behavior with other things that called Dispose directly.